### PR TITLE
Check length of mail address

### PR DESCRIPTION
**Pull request by Udera**
Check if mail address is RFC conform.
partly fixes https://github.com/vexim/vexim2/issues/108

### CODE REVIEW

File: vexim/config/functions.php [hunk 1]
```diff
@@ -43,6 +43,24 @@ function check_user_exists($dbh,$localpart,$domain_id,$page)
...
+     *
+     * @param  string  $localpart
+     * @param  string  $domain
+     * @param  string  $page       page to return to
+     */
+    function check_mail_address($localpart,$domain,$page)
+    {
+      if ((strlen($localpart)+strlen($domain)>253) || strlen($localpart)>64)
+        {
+            header ("Location: $page?addresstoolong=$localpart");
```
Comment (rimas-kudelis) [inline vexim/config/functions.php]: This doesn't seem to account for the possibility of vexim being hosted in a subdirectory.

File: vexim/config/functions.php [hunk 2]
```diff
@@ -43,6 +43,24 @@ function check_user_exists($dbh,$localpart,$domain_id,$page)
         }
     }
 
+    /**
+     * Check if mail address is compliant with RFC 3696.
+     *
+     * Queries database $dbh, and redirects to the $page is the user already
+     * exists.
```
Comment (rimas-kudelis) [inline vexim/config/functions.php]: ORLY?

File: vexim/adminaliasaddsubmit.php [hunk 3]
```diff
@@ -43,6 +43,11 @@
     die;
   }
 
+  # check_mail_address() will die if the localpart (>64 char) or the total address (>254 char) is too long
```
Comment (rimas-kudelis) [inline vexim/adminaliasaddsubmit.php]: This comment seems unnecessary, especially considering that you haven't added it in other places.

File: vexim/config/functions.php [hunk 4]
```diff
@@ -43,6 +43,24 @@ function check_user_exists($dbh,$localpart,$domain_id,$page)
         }
     }
 
+    /**
+     * Check if mail address is compliant with RFC 3696.
```
Comment (rimas-kudelis) [inline vexim/config/functions.php]: It's probably worth adding specifying the limits (254 and 64) here.

File: vexim/config/functions.php [hunk 5]
```diff
@@ -43,6 +43,24 @@ function check_user_exists($dbh,$localpart,$domain_id,$page)
...
+     *
+     * Queries database $dbh, and redirects to the $page is the user already
+     * exists.
+     *
+     * @param  string  $localpart
+     * @param  string  $domain
+     * @param  string  $page       page to return to
```
Comment (rimas-kudelis) [inline vexim/config/functions.php]: This should say "page to return to in case of failure".

File: vexim/config/functions.php (hunk 1 above)
Comment (Udera) [inline vexim/config/functions.php]: This is the same trick we use in `check_user_exists()`. All accessible pages within vexim reside in the same directory, a simple relative link should be fine. Or do I misunderstand something?

**Comment by Udera** [comment]
Another try :D

File: vexim/config/functions.php (hunk 1 above)
Comment (rimas-kudelis) [inline vexim/config/functions.php]: Uhm, I was somehow under impression that this redirect URL would be treated as absolute, not relative. If this is not the case, I retract my comment.

**Pull request merged**